_drafts/firmware-code-size-2.md

```diff
@@ -0,0 +1,303 @@
+---
+title: "Shrink your Firmware with GCC"
+description: "A walk through GCC configuration options to optimize firmware code size"
+author: francois
+---
+
+## Introduction
+This is the second post in our Firmware Code Size Optimization series. Last time, we talked about measuring code size as a precondition to improving it. Now we’ll dive leap from planning into action!
+
+<!-- excerpt start -->
+In this post, we will review compiler options that we can use to reduce firmware code size. We will focus on `arm-none-eabi-gcc`, the GCC compiler used for ARM-based microcontroller, though most of the compile-time flags we will cover are available in other GCC flavors as well as in Clang.
+<!-- excerpt end -->
+
+You would think that a single flag could be used for “make this as small as possible”, but unfortunately it isn’t so. Instead, code size optimization involves complicated trade-offs that must be considered on a case by case basis.
+
+_Like Interrupt? [Subscribe](http://eepurl.com/gpRedv) to get our latest posts straight to your mailbox_
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
+## Setting the stage
+In order to get measurable improvements in code size, we need a reasonably large code-base to begin with. To that end, we’ll use example code from [ChibiOS](http://chibios.org/dokuwiki/doku.php), a free and open source RTOS. Specifically, we’ll use their FatFS+USB example for STM32F1 MCUs.
```

Comment:
nit: 

```
ChibiOS[^1]

[...]
[^1]: [ChibiOS](http://chibios.org/dokuwiki/doku.php)
```

might be more preferable for links in general but not a big deal

Reply:
Done